src/game/tictactoe/ttt_game.py

```diff
@@ -0,0 +1,333 @@
+# src/game/chess.py
+import math
+import chess
+import chess.svg
+import random
+from src.game.base_game import BaseGame
+from src.utility.ttt_extraction import available_moves, make_move, undo_move, check_winner, minimax
+from itertools import permutations
+import json
+
+# *************** Game State Based Evaluation ***************
+# Note: pure square based evaluation is available below.
+weight_labels = [
+    # board states
+    "corners-controlled",
+    "middle-controlled",
+    "two-in-a-row",
+
+    # move types
+    "winning-move",
+    "blocking-win",
+    "forking-move", # creating a fork (3 adjacent corners)
+    "blocking-fork", # block opponent fork creation
+    "creating-fork-for-next-move", # two adjacent corners, create opportunity to create a fork next turn
+]
+weight_bounds = [
+    (-10, 10), # corners-controlled
+    (-10, 10), # middle-controlled
+    (-10, 10), # two-in-a-row
+    (-10, 10), # winning-move
+    (-10, 10), # blocking-win
+    (-10, 10), # forking-move
+    (-10, 10), # blocking-fork
+    (-10, 10), # creating-fork-for-next-move
+]
+
+class tttGame(BaseGame):
+    def __init__(self, meta):
+        board, moves_and_scores = meta
+
+        self.board = board
+        self.moves_and_scores = moves_and_scores
+        self.initialize_random_weights()
+        self.rank_moves()
+        
+
+    # Random starting genes for the chromosome based on lower and upper bounds
+    def initialize_random_weights(self):
+        self.weights = [random.uniform(float(lower), float(upper)) for lower, upper in weight_bounds]
+
+    def get_board_data(self):
+        return [self.board, self.moves_and_scores]
+
+    def rank_move(self, move):
+        rank = 1
+        for v in self.moves_and_scores:
+            if v[0] == move:
+                return (rank, len(list(self.moves_and_scores)))
+            rank += 1
+        
+        return (rank, len(list(self.moves_and_scores)))
+
+    def get_weights(self):
+        return self.weights
+    
+    def get_best_move(self):
+        best_move, best_score = self.determine_best_move_from_weights()
+        return best_move
+    
+    def rank_moves(self):
+        self.moves_and_scores = sorted(self.moves_and_scores, key=lambda x: x[1], reverse=True)
```

Comment:
you didn't change this to a proper data structure... this is innefficient. Normally I would say change it but tic tac toe is not very intensive so it's not that big a deal.